File: CS262/4.breakingupstrings.py
def rsplit(string,pos=0):
    cache = string.find(" ", pos)
    if pos < len(string) and cache >=0:
        print cache
        pos = cache + 1
        rsplit(string,pos)

string = "Jose Mariz Melo Formiga Viana"
print "WHITE SPACES ON : "+string
rsplit(string)

File: CS262/4.breakingUpStrings.cpp
#include <iostream>
#include <string>
using namespace std;

int rsplit(string str, int pos=0){
    size_t found;
    found=str.find(" ", pos);
    int cache = int(found);
    if( pos < str.length() && cache >=0 ){
        cout << cache << endl;
        pos = cache + 1;
        rsplit(str, pos);
    }//if
}//RSPLIT()

int main(){
    rsplit("jose mariz melo formiga viana");
}//MAIN()